src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+package parser;
+
+import java.util.Scanner;
+
+import storage.Storage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+import task.TaskList;
+
+/**
+ * Read in user input and execute relative commands.
+ */
+
+public class Parser {
+    protected TaskList todoList;
+    protected Storage storage;
+
+    public Parser(TaskList todoList, Storage storage) {
+        this.todoList = todoList;
+        this.storage = storage;
+    }
+
+    /**
+     * Recognize keywords from the user input such as "bye" and "list".
+     * If not, pass it to further cases of commands.
+     */
+
+    public void parse() {
+        boolean isExit = false;
+        Scanner scanner = new Scanner(System.in);
+        while ((!isExit) && scanner.hasNextLine()) {
+            String userInput = scanner.nextLine();
+            if (userInput.equals("bye")) {
+                isExit = true;
+            } else if (userInput.equals("list")) {
+                this.todoList.printList();
+            } else if (isMarkTask(userInput)) {
+                this.todoList.changeMarkingOfTask(userInput, storage);
+            } else if (isDeleteTask(userInput)) {
+                this.todoList.deleteTask(userInput, storage);
+            } else if (isFindTask(userInput)) {
+                this.todoList.findTask(userInput, storage);
+            } else {
+                this.echo(userInput);
+            }
+        }
+        scanner.close();
+    }
+
+    /**
+     * Handle the three cases when the user input is "todo", "deadline" or "event".
+     * 
+     * @param userInput the input from the user.
+     */
+
+    public void echo(String userInput) {
```

Comment:
This method is too long, try refactoring it by extracting out some parts into a new method